opensrp-reveal/src/test/java/org/smartregister/reveal/presenter/TaskRegisterFragmentPresenterTest.java

```diff
@@ -0,0 +1,297 @@
+        verify(view).showProgressView();
+        verify(interactor).findTasks(mainConditionCaptor.capture(), myLocationCaptor.capture(), operationalAreaCenterCaptor.capture());
+        assertEquals(mainCondition, mainConditionCaptor.getValue().first);
+        assertNull(mainConditionCaptor.getValue().second[0]);
+        assertNull(mainConditionCaptor.getValue().second[1]);
+
+        assertEquals(location, myLocationCaptor.getValue());
+
+        assertNull(operationalAreaCenterCaptor.getValue());
+    }
+
+    @Test
+    public void testInitializeQueriesUsesCorrectParams() {
+        String campaignId = UUID.randomUUID().toString();
+        when(preferencesUtil.getCurrentCampaignId()).thenReturn(campaignId);
+        when(preferencesUtil.getCurrentOperationalArea()).thenReturn("MTI_84");
+        String mainCondition = "group_id = ? AND campaign_id = ?";
+        Whitebox.setInternalState(presenter, "visibleColumns", visibleColumns);
+        presenter.initializeQueries(mainCondition);
+        verify(view).initializeAdapter(eq(visibleColumns));
+        verify(view).showProgressView();
+        verify(interactor).findTasks(mainConditionCaptor.capture(), myLocationCaptor.capture(), operationalAreaCenterCaptor.capture());
+        assertEquals(mainCondition, mainConditionCaptor.getValue().first);
+        assertEquals(operationalArea.getId(), mainConditionCaptor.getValue().second[0]);
+        assertEquals(campaignId, mainConditionCaptor.getValue().second[1]);
+
+        assertEquals(location, myLocationCaptor.getValue());
+
+        assertEquals(-14.152197, operationalAreaCenterCaptor.getValue().getLatitude(), 0.0001);
+        assertEquals(32.643570, operationalAreaCenterCaptor.getValue().getLongitude(), 0.0001);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testStartSync() {
+        presenter.startSync();
+    }
+
+    @Test
+    public void testOnTasksFoundWithNullTasks() {
+        presenter.onTasksFound(null, 0);
+        verify(view).displayNotification(R.string.fetching_structure_title,
+                R.string.fetch_location_and_structures_failed, preferencesUtil.getCurrentOperationalArea());
+        verify(view).setTaskDetails(eq(new ArrayList<>()));
+        verify(view).hideProgressDialog();
+        verify(view).hideProgressView();
+        verify(view).setTotalTasks(eq(0));
+        verifyNoMoreInteractions(view);
+
+    }
+
+    @Test
+    public void testOnTasksFoundWithEmptyTasks() {
+        presenter.onTasksFound(new ArrayList<>(), 0);
+        verify(view).displayNotification(eq(R.string.fetching_structure_title), eq(R.string.no_structures_found));
+        verify(view).setTaskDetails(eq(new ArrayList<>()));
+        verify(view).hideProgressDialog();
+        verify(view).hideProgressView();
+        verify(view).setTotalTasks(eq(0));
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void testOnTasksFoundWithTasks() {
```

Comment:
In future, can we use one format for the test method names? If such a test method failed, it's not clear from the name what the expected result should be so as to quickly fix it.

Reply:
I personally prefer `unitOfWorkShouldExpectedBehavior[whenStateUnderTest]`

Reply:
This can be agreed upon however I have adopted android best practices as documented on the below link. What is missing is just using the comma for specif use case being tested where am using camel case. This should be easy to follow if using android guidelines

[https://source.android.com/setup/contribute/code-style#javatests-style-rules](https://source.android.com/setup/contribute/code-style#javatests-style-rules)